### DISCUSSION TITLE
Detecting scroll end

### DISCUSSION THREAD

**Discussion by itishermann** [icflorescu/mantine-datatable#209] [Q&A]
Hi everyone,
Can someone tell me how to detect the scroll end when the Datatable is scrollable ?
i tried adding an event listener to the body ref but it didn't work

**Comment by icflorescu** [comment]
Hey @itishermann,

It's not possible in the current version, but I think it wouldn't be too difficult for me to expose the underlying [ScrollArea's `onScrollPositionChange`](https://mantine.dev/core/scroll-area/#subscribe-to-scroll-position-changes).

What exactly are you trying to achieve?

I'm trying to understand if this would be a potentially useful feature for other people.

Thanks!

**Reply by itishermann** [reply]
hi, i'm trying to achieve an inifinte scroll, at the company i'm working for, we use cursor based pagination so the included pagination doesn't really fits.

**Reply by icflorescu** [reply]
I see.
In this case simply exposing the ScrollArea's `onScrollPositionChange` wouldn't be enought.
I think it would be helpful if Mantine DataTable would expose an `onScrollPositionChange` event that gives you the current scroll position **and** the underlying ScrollArea's `scrollHeight` and `scrollWidth`, so you can establish whether you're nearing a "scroll end threshold" and trigger the data loading, am I right?...

**Reply by itishermann** [reply]
exactly, that would be perfect

**Reply by icflorescu** [reply]
Or better yet, to make life easier, the component could expose these granular events: `onScrollToTop`, `onScrollToBottom`, `onScrollToLeft`, `onScrollToRight`.
This way you could simply use `onScrollToBottom` to trigger loading more data.
What do you think?

**Comment by icflorescu** [comment]
Starting with `v2.1.0` you can use the `onScrollToBottom` event 🎉
See this example: https://icflorescu.github.io/mantine-datatable/examples/infinite-scrolling
